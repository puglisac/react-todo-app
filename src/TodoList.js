import React, {useState} from 'react';
import Todo from './Todo';
import NewTodoForm from './NewTodoForm';
import { v4 as uuidv4 } from 'uuid';
// import './TodoList.css'

const TodoList = ()=> {
    const [todos, setTodos]=useState([]);

    const addTodo=( text )=>{
        const id = uuidv4();
        setTodos(todos=>[...todos,{id, text}]);

    }

    const deleteTodo=(id)=>{
        const idx = todos.findIndex((t)=>t.id===id);
        todos.splice(idx,1);
        setTodos(todos=>[...todos]);
    }

    return(
        <>

        <NewTodoForm addTodo={addTodo}/>
        <div className="TodoList">
            <h2>Todos</h2>
            <ul>
        {todos.map(t=> (<Todo key={t.id} 
            text={t.text} 
            id={t.id}
            deleteTodo={deleteTodo}/>))}
            </ul>
            </div>
        
        </>
    )
}
export default TodoList;
