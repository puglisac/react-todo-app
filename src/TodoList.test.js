import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TodoList from './TodoList';

it("renders without crashing", function() {
    render(<TodoList />);
  });

  it("matches snapshot", function() {
    const {asFragment} = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
  });

it("can and and delete a todo", function() {
    const { getByLabelText, queryByText, getByText } = render(<TodoList />);
  
    const input = getByLabelText("Todo:");
    const submitBtn = getByText("Add Todo!")
    expect(queryByText("todo")).not.toBeInTheDocument();
  
    // fill out the form
    fireEvent.change(input, { target: { value: "todo" }});
    fireEvent.click(submitBtn);
  
    expect(queryByText("todo")).toBeInTheDocument();
  

    const deleteBtn = getByText("X");
    fireEvent.click(deleteBtn);
    expect(queryByText("todo")).not.toBeInTheDocument();  
  });