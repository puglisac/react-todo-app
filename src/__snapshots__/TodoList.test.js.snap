// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="text"
    >
      Todo:
    </label>
    <input
      id="text"
      name="text"
      value=""
    />
    <button>
      Add Todo!
    </button>
  </form>
  <div
    class="TodoList"
  >
    <h2>
      Todos
    </h2>
    <ul />
  </div>
</DocumentFragment>
`;
